python/delta_sharing/tests/test_rest_client.py

```diff
@@ -24,10 +27,54 @@
     Share,
     Table,
 )
-from delta_sharing.rest_client import DataSharingRestClient
+from delta_sharing.rest_client import DataSharingRestClient, retry_with_exponential_backoff
 from delta_sharing.tests.conftest import ENABLE_INTEGRATION, SKIP_MESSAGE
 
 
+def test_retry(rest_client: DataSharingRestClient):
+    class TestWrapper(DataSharingRestClient):
+        def __init__(self):
+            # inherit from DataSharingRestClient to make sure all the helper methods are the same
+            super().__init__(rest_client._profile)
+            self.sleeps = []
+            self._sleeper = self.sleeps.append
+
+            error = HTTPError()
+            response = Response()
+            response.status_code = 429
+            error.response = response
+            self.error = error
+
+        @retry_with_exponential_backoff
+        def success(self):
+            return True
+
+        @retry_with_exponential_backoff
+        def all_fail(self):
+            raise self.error
+
+        @retry_with_exponential_backoff
+        def fail_before_success(self):
+            if len(self.sleeps) < 4:
+                raise self.error
+            else:
+                return True
+
+    wrapper = TestWrapper()
+    assert wrapper.success()
```

Comment:
nit: we might want to check `wrapper.sleeps` here as well to confirm it succeeds without any error.

Reply:
Added :)